kernel/src/process/signal/mod.rs

```diff
@@ -93,8 +94,7 @@ pub fn handle_pending_signal(
             }
 
             drop(sig_dispositions);
-
-            handle_user_signal(
+            if let Err(e) = handle_user_signal(
```

Comment:
So after this change, it seems that `handle_pending_signal` will no longer return `Err(_)`. So the signature can be changed to return `()`.